todo_app/templates/index.html

```diff
@@ -41,9 +50,15 @@ <h4>To-Do</h4>
         <div class="row">
           <div class="col-sm">
             <b>{{ item.title }}</b> - {{ item.description }}
-            <a href="{{ url_for('do_item', item_id=item.id) }}" class="btn btn-primary btn-sm" role="button">Mark as Doing</a>
-            <a href="{{ url_for('complete_item', item_id=item.id) }}" class="btn btn-success btn-sm" role="button">Mark as Done</a>
-            <a href="{{ url_for('remove_item', item_id=item.id) }}" class="btn btn-danger btn-sm" role="button">Delete</a>
+            {% if current_user.id in WRITER_LIST or not current_user.is_authenticated %}
```

Comment:
You have this same check quite a few times - it's potentially something we could add as a property on the ViewModel to make it easier, e.g. as `view_model.can_see_write_controls`

Reply:
Adding to the next update